user-impl/src/main/java/com/example/auction/user/impl/UserServiceImpl.java

```diff
@@ -26,37 +26,35 @@
     public UserServiceImpl(PersistentEntityRegistry registry, UserRepository userRepository) {
         this.registry = registry;
         this.userRepository = userRepository;
-
         registry.register(PUserEntity.class);
     }
 
 
     @Override
     public ServiceCall<UserRegistration, User> createUser() {
         return user -> {
-            UUID uuid = UUID.randomUUID();
+            UUID uuid = getUUIDFromEmail(user.getEmail());
             String password = PUserEntity.hashPassword(user.getPassword());
-            PUser createdUser = new PUser(uuid,  user.getName(), user.getEmail(), password);
+            PUser createdUser = new PUser(uuid, user.getName(), user.getEmail(), password);
             return entityRef(uuid)
                 .ask(new PUserCommand.CreatePUser(user.getName(), user.getEmail(), password))
                 .thenApply(done -> Mappers.toApi(Optional.ofNullable(createdUser)));
         };
     }
 
+    private UUID getUUIDFromEmail(String email) {
+        return UUID.nameUUIDFromBytes(email.toLowerCase().getBytes());
+    }
```

Comment:
This is the major change on this PR: `userId` is derived from the `email`.  This is a first step into making the `userId` the actual user `email` but that would have taken me a bit longer.

Reply:
It's a good thing we haven't implemented a way to update your email address! 😆